File: server/viewmodels/comment.js
const UserViewModel = require('./user')

module.exports = class CommentViewModel {
  constructor () {
    this.id = 0
    this.issueId = 0
    this.body = ''
    this.author = null
    this.updatedAt = new Date()
    this.createdAt = new Date()
  }

  static createFromJira (obj) {
    let result
    if (obj) {
      result = new CommentViewModel()
      result.id = obj.id
      result.body = obj.body
      result.author = UserViewModel.createFromJira(obj.author)
      result.createdAt = obj.created
      result.updatedAt = obj.updated
    }
    return result
  }

  static createFromLocal (obj) {
    let result
    if (obj) {
      result = new CommentViewModel()
      result.id = obj.id
      result.body = obj.body
      result.author = UserViewModel.createFromLocal(obj.author)
      result.createdAt = obj.created
      result.updatedAt = obj.updated
    }
    return result
  }
}

File: server/model/comment.py
from ._base import BaseModel
from .user import User


class Comment(BaseModel):
    def __init__(self, id_=0, body='', updated='', created=''):
        super().__init__(id_)
        self.author = User()
        self.body = body
        self.updatedAt = updated
        self.createdAt = created

    @classmethod
    def from_db_dict(cls, obj):
        comment = cls(
            obj['id'],
            obj['body'],
            obj['updatedAt'],
            obj['createdAt'],
        )
        comment.author = User(
            obj['user_id'],
            obj['user_externalId'],
            obj['user_username'],
            obj['user_displayName'],
            obj['user_avatar'],
        )
        return comment

    def to_viewmodel(self):
        return {
            'id': self.id,
            'body': self.body,
            'author': self.author.to_viewmodel(),
            'createdAt': self.createdAt,
            'updatedAt': self.updatedAt,
        }
